realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -367,7 +372,13 @@ private void completedAsyncRealmResults(QueryUpdateTask.Result result) {
                     if (!realmResults.isLoaded()) {
                         RealmLog.d("[COMPLETED_ASYNC_REALM_RESULTS "+ weakRealmResults + "] , realm:"+ HandlerController.this + " same versions, using results (RealmResults is not loaded)");
                         // swap pointer
-                        realmResults.swapTableViewPointer(result.updatedTableViews.get(weakRealmResults));
+                        try {
+                            realmResults.swapTableViewPointer(result.updatedTableViews.get(weakRealmResults));
+                        } catch (BadVersionException e) {
```

Comment:
If the caller advanced then we won't hit this code path at all because that was already checked for. `begin_read()` is the only place where it makes sense to catch a BadVersionException, as everywhere else it the precondition can be checked first.